### PR TITLE
added SeverityPieChartTest

### PR DESCRIPTION
**Pull request by MatthiasHer**

### CODE REVIEW

**Comment by MatthiasHer** [comment]
@TanjaRo1

File: src/test/java/io/jenkins/plugins/analysis/core/charts/SeverityPieChartTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ *
+ * @author Matthias Herpers
+ */
+class SeverityPieChartTest {
+
+    /**
+     *  Verifies that a SeverityPieChart with empty report shows only predefined severities without Severity.ERROR.
+     */
+    @Test
+    void buildTestWithEmptyReport() {
```
Comment (TanjaRo1) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/SeverityPieChartTest.java]: Better use a more meaningful method name starting with should eg. ShouldCreateChartFromEmptyReport (cf. [https://github.com/uhafner/codingstyle/blob/master/doc/Testen.md](url)) (applies also to the other method names)

File: src/test/java/io/jenkins/plugins/analysis/core/charts/SeverityPieChartTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+        SeverityPieChart severityPieChart = new SeverityPieChart();
+        int numberOfDefaultPieData = Severity.getPredefinedValues().size() - 1; // Without Severity.ERROR
+        //Create Report
```
Comment (TanjaRo1) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/SeverityPieChartTest.java]: Do not write comments that describe the code, as these will not survive a refactoring. (applies also to the other comments inside your methods)

File: src/test/java/io/jenkins/plugins/analysis/core/charts/SeverityPieChartTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import static org.assertj.core.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+/**
+ * Tests the class {@link SeverityPieChart}.
+ *
+ * @author Matthias Herpers
+ */
+class SeverityPieChartTest {
+
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/SeverityPieChartTest.java]: No empty lines

File: src/test/java/io/jenkins/plugins/analysis/core/charts/SeverityPieChartTest.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    /**
+     *  Verifies that a SeverityPieChart with empty report shows only predefined severities without Severity.ERROR.
+     */
+    @Test
+    void shouldCreateChartFromEmptyReport() {
+
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/SeverityPieChartTest.java]: Empty lines should structure your code,see [coding style](https://github.com/uhafner/codingstyle/blob/master/doc/Formatierung.md#leerzeilen)

File: src/test/java/io/jenkins/plugins/analysis/core/charts/SeverityPieChartTest.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        SeverityPieChart severityPieChart = new SeverityPieChart();
+        int numberOfDefaultPieData = Severity.getPredefinedValues().size() - 1; // Without Severity.ERROR
+        Report report = createReport(0, 0, 0, 0);
+        PieChartModel pieChartModel = severityPieChart.create(report);
+        List<PieData> data = pieChartModel.getData();
+        for (PieData severity : data) {
+            assertThat(Severity.getPredefinedValues().contains(severity));
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/SeverityPieChartTest.java]: This does not check anything, use
```
assertThat(actual).isEqualTo(expected);
assertThat(actualBoolean).isTrue();
```
See https://joel-costigliola.github.io/assertj/assertj-core-quick-start.html.

See also https://ci.jenkins.io/job/Plugins/job/warnings-ng-plugin/job/PR-47/6/spotbugs/new/

File: src/test/java/io/jenkins/plugins/analysis/core/charts/SeverityPieChartTest.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        }
+
+        assertThat(data.size()).isEqualTo(numberOfDefaultPieData);
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/SeverityPieChartTest.java]: Check the actual values as well

File: src/test/java/io/jenkins/plugins/analysis/core/charts/SeverityPieChartTest.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        int numberOfDefaultPieData = Severity.getPredefinedValues().size(); // With Severity.ERROR
+        Report report = createReport(0, 0, 0, 1);
+        PieChartModel pieChartModel = severityPieChart.create(report);
+        List<PieData> data = pieChartModel.getData();
+        for (PieData severity : data) {
+            assertThat(Severity.getPredefinedValues().contains(severity));
+        }
+
+        assertThat(data.size()).isEqualTo(numberOfDefaultPieData);
+    }
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/SeverityPieChartTest.java]: see comments above

File: src/test/java/io/jenkins/plugins/analysis/core/charts/SeverityPieChartTest.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        int numberOfDefaultPieData = Severity.getPredefinedValues().size(); // With Severity.ERROR
+        Report report = createReport(1, 1, 1, 1);
+        PieChartModel pieChartModel = severityPieChart.create(report);
+        List<PieData> data = pieChartModel.getData();
+        for (PieData severity : data) {
+            assertThat(Severity.getPredefinedValues().contains(severity));
+        }
+
+        assertThat(data.size()).isEqualTo(numberOfDefaultPieData);
+    }
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/core/charts/SeverityPieChartTest.java]: see comments above

**Pull request merged**